Sources/StreamChat/APIClient/Endpoints/Payloads/ThreadListPayload.swift

```diff
@@ -0,0 +1,88 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+struct ThreadListPayload: Decodable {
```

Comment:
Why don't we also add the networking part in this PR?

Reply:
It is just easier to review if the PRs are split into smaller ones